webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -0,0 +1,210 @@
+     return(   
+      <div className="container"  align="center" >
+      <p className="h5 pt-5">We are pleased to offer you a place at the Deep Learning Indaba 2019. Please see the details of this offer below </p>
+
+      <form class="pt-5 ">
+        <p class="font-weight-bold">You have been accepted as a  {userProfile!=null ? userProfile.user_category: "<Category>"} </p>
+        <p class="font-weight-bold">Your status</p>
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Travel Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.travel_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Accommodation Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.accomodation_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row pb-5 ">
+          <div class="col-8 font-weight-bold">
+            Payment Required:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.payment_required
+          : "Not available"}
+          </div>
+        </div>
+        <p class="font-weight-bold">Please accept or reject this offer by {offerList!=null ? offerList.expiry_date : "<Expiry Date>"} </p>
+        <div class="row">
+         <div class="col">
+          <button type="button" class="btn btn-danger" id="reject"onClick={
+            ()=>{
+              this.setState({
+                rejected:true
+              });
+              this.buttonSubmit();
+            }
+          } >
+              Reject
+          </button>
+          </div>
+          <div class="col">
+          <button type="button" class="btn btn-success" id="accept" onClick={()=>{
+              this.setState({
+                accepted:true,
+                rejected:false,
+              });
+              this.buttonSubmit();
+          }}>
+              Accept
+          </button>
+          </div>
+        </div> 
+
+        <div class="form-group">
+         { this.state.rejected ? 
+          <div class="form-group mr-5  ml-5 pt-5" >
+          <textarea class="form-control pr-5 pl-10" id="exampleFormControlTextarea3" onChange={()=>{this.handleChange(this.rejected_reason)}}  placeholder="Enter rejection message"></textarea>
+          <button type="button" class="btn" id="apply" onClick={
+            ()=>{
+              this.setState({
+                rejected:false
+              });
+              this.buttonSubmit();
+            }
+            }>apply</button>
+        </div>
+        :""}
+        </div>
+      
+      </form>
+  </div>
+      )}
+
+    componentDidMount() {
+          this.getOfferList();
+          console.log("<<< List >>>>",this.state.offerList)
+          profileService.getProfilesList(DEFAULT_EVENT_ID)
```

Comment:
This method returns a **list** of profiles - ie everyone who applied for the event! We only need one piece of information (the user_category) from the user whose offer is being viewed. Either use the UserProfile controller (`/api/v1/userprofile`) or better, update the Offer controller to return the user_category along with the offer.

Reply:
@LindanyRetroRabbit I'm going to suggest returning user_category along with the offer